plane/src/typed_unix_socket/server.rs

```diff
@@ -80,43 +70,39 @@ where
             event_tx,
             request_tx,
             response_tx,
-            _phantom: PhantomData,
         })
     }
 
-    pub fn subscribe_events(&self) -> broadcast::Receiver<ClientMessageType> {
+    pub fn subscribe_events(&self) -> broadcast::Receiver<MessageToServer> {
         self.event_tx.subscribe()
     }
 
-    pub fn subscribe_requests(&self) -> broadcast::Receiver<IDedMessage<RequestType>> {
+    pub fn subscribe_requests(&self) -> broadcast::Receiver<WrappedMessage<MessageToServer>> {
         self.request_tx.subscribe()
     }
 
-    pub async fn send_response(
-        &self,
-        request: IDedMessage<RequestType>,
-        response: ResponseType,
-    ) -> Result<(), Error> {
-        let response_msg =
-            WrappedServerMessageType::<ResponseType, ServerMessageType>::Response(IDedMessage {
-                id: request.id,
-                message: response,
-            });
-
+    pub async fn send_response(&self, id: String, response: MessageToClient) -> Result<(), Error> {
```

Comment:
One thing I was intentional about with the API design is I'd like the details of the ID to be as hidden as possible from the calling code. As such, it'd be nice if `send_response` could accept the request and response like before, and internally take care of extracting/reusing the ID.